bigint/src/bigint.rs

```diff
@@ -926,6 +926,56 @@ impl BigInt {
         BigInt::from_biguint(sign, BigUint::new(digits))
     }
 
+    /// Creates and initializes a `BigInt` from an array of bytes in two's complement.
+    ///
+    /// The digits are in little-endian base 2^8.
+    #[inline]
+    pub fn from_twos_complement_bytes_le(mut digits: Vec<u8>) -> BigInt {
+        let sign = if digits.iter().all(Zero::is_zero) {
+            Sign::NoSign
+        } else {
+            digits.iter().rev().next()
+                .map(|v| if *v > 127 {
+                    Sign::Minus
+                } else {
+                    Sign::Plus
+                })
+                // we have at least one non-zero digit
+                .unwrap()
+        };
```

Comment:
I'd just `match digits.last()` here, with `None => BigInt::zero()`, and let `from_biguint` take care of normalizing longer zero sequences.